examples/consistency_distillation/train_lcm_distill_lora_sd_wds.py

```diff
@@ -360,18 +360,42 @@ def scalings_for_boundary_conditions(timestep, sigma_data=0.5, timestep_scaling=
 
 # Compare LCMScheduler.step, Step 4
 def predicted_origin(model_output, timesteps, sample, prediction_type, alphas, sigmas):
+    alphas = extract_into_tensor(alphas, timesteps, sample.shape)
+    sigmas = extract_into_tensor(sigmas, timesteps, sample.shape)
     if prediction_type == "epsilon":
-        sigmas = extract_into_tensor(sigmas, timesteps, sample.shape)
-        alphas = extract_into_tensor(alphas, timesteps, sample.shape)
         pred_x_0 = (sample - sigmas * model_output) / alphas
+    elif prediction_type == "sample":
+        pred_x_0 = model_output
     elif prediction_type == "v_prediction":
-        pred_x_0 = alphas[timesteps] * sample - sigmas[timesteps] * model_output
+        pred_x_0 = alphas * sample - sigmas * model_output
     else:
-        raise ValueError(f"Prediction type {prediction_type} currently not supported.")
+        raise ValueError(
+            f"Prediction type {prediction_type} is not supported; currently, `epsilon`, `sample`, and `v_prediction`"
+            f" are supported."
+        )
 
     return pred_x_0
 
 
+# Based on step 4 in DDIMScheduler.step
+def predicted_source_noise(model_output, timesteps, sample, prediction_type, alphas, sigmas):
```

Comment:
what does `source` refer to here ?

Reply:
Here `source` refers to the source noise $`\epsilon`$ ($`\epsilon_t`$ in Equation (5) of the [DDIM paper](https://arxiv.org/pdf/2010.02502.pdf)) that the denoising model tries to predict when `prediction_type == "epsilon"`.

Would it be more clear is this was named something like `predicted_epsilon` (which uses terminology closer to that of `DDIMScheduler`)?

Reply:
Saw https://github.com/huggingface/diffusers/pull/6145#discussion_r1427694793 below, `get_predicted_noise` works for me